src/AspNet.Security.OAuth.BungieNet/BungieNetAuthenticationHandler.cs

```diff
@@ -0,0 +1,128 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.BungieNet;
+
+public partial class BungieNetAuthenticationHandler : OAuthHandler<BungieNetAuthenticationOptions>
+{
+    public BungieNetAuthenticationHandler(
+        [NotNull] IOptionsMonitor<BungieNetAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder,
+        [NotNull] ISystemClock clock)
+        : base(options, logger, encoder, clock)
+    {
+    }
+
+    protected override async Task<AuthenticationTicket> CreateTicketAsync(
+        [NotNull] ClaimsIdentity identity,
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        var membershipId = tokens.Response?.RootElement.GetString("membership_id");
+        var userInfoEndpoint = string.Format(CultureInfo.InvariantCulture, Options.UserInformationEndpoint, membershipId);
```

Comment:
Does the membership ID need to be URI-escaped?

Reply:
No, its number